src/test/scala/uk/gov/nationalarchives/tdr/api/db/repository/FileRepositorySpec.scala

```diff
@@ -0,0 +1,110 @@
+    val db = DbConnection.db
+    val fileRepository = new FileRepository(db)
+    val consignmentId = UUID.fromString("64456c78-49bb-4bff-85c8-2fff053b9f8e")
+
+    TestUtils.createConsignment(consignmentId, userId)
+    TestUtils.createFile(UUID.fromString("2c2dedf5-56a1-497f-9f8a-19102739a055"), consignmentId)
+    TestUtils.createFile(UUID.fromString("c2d5c569-0fc4-4688-9523-157c4028f1b0"), consignmentId)
+    TestUtils.createFile(UUID.fromString("317c7084-d3d4-435b-acd2-cfb317793843"), consignmentId)
+//  We have created files, but not added any AVmetadata for those files to the AVMetadata repository.
+//  Thus, calling the countProcessedFilesInConsignment method should return 0.
+
+    val avMetadataFiles = fileRepository.countProcessedFilesInConsignment(consignmentId).futureValue
+
+    avMetadataFiles shouldBe 0
+  }
+
+  "countProcessedFilesInConsignment" should "return the number of AVmetadata for files in a specified consignment" in {
+    val db = DbConnection.db
+    val fileRepository = new FileRepository(db)
+    val consignmentOne = "049a11d7-06f5-4b11-b786-640000de76e1"
+    val consignmentTwo = "dc2acd2c-3370-44fe-aeae-16aee30dc41f"
+    val fileOneId = "5a102380-25ec-4881-a484-87d56fe9a0b3"
+    val fileTwoId = "f616272d-4d80-44cf-90de-996c3984847c"
+    val fileThreeId = "40c67a87-70c0-4424-b934-a395459ddbe0"
+
+//  Need to create a consignment with files in it
+    TestUtils.createConsignment(UUID.fromString(consignmentOne), userId)
+    TestUtils.createConsignment(UUID.fromString(consignmentTwo), userId)
+    TestUtils.createFile(UUID.fromString(fileOneId), UUID.fromString(consignmentOne))
+    TestUtils.createFile(UUID.fromString(fileTwoId), UUID.fromString(consignmentOne))
+    TestUtils.createFile(UUID.fromString(fileThreeId), UUID.fromString(consignmentTwo))
+
+//  Then need to add data to the AVMetadata repository for these files
+    TestUtils.addAntivirusMetadata(fileOneId, consignmentOne)
+    TestUtils.addAntivirusMetadata(fileTwoId, consignmentOne)
+    TestUtils.addAntivirusMetadata(fileThreeId, consignmentTwo)
+
+    val avMetadataFiles = fileRepository.countProcessedFilesInConsignment(UUID.fromString(consignmentOne)).futureValue
+
+    avMetadataFiles shouldBe 2
+  }
+
+  "countProcessedFilesInConsignment" should "return number of AVmetadata rows with repetitive data filtered out" in {
+    val db = DbConnection.db
+    val fileRepository = new FileRepository(db)
+    val consignmentId = "c6f78fef-704a-46a8-82c0-afa465199e65"
+    val fileOneId = "20e0676a-f0a1-4051-9540-e7df1344ac10"
+    val fileTwoId = "b5111f11-4dca-4f92-8239-505da567b9df"
+    var count = 0
+
+    TestUtils.createConsignment(UUID.fromString(consignmentId), userId)
+    TestUtils.createFile(UUID.fromString(fileOneId), UUID.fromString(consignmentId))
+    TestUtils.createFile(UUID.fromString(fileTwoId), UUID.fromString(consignmentId))
+
+    while (count < 7) {
```

Comment:
A slightly more idiomatic way of doing this in Scala would be:

```
(1 to 7).foreach { _ => TestUtils.addAntivirusMetadata(fileOneId, consignmentId) }
```

(I hope my syntax is right there - I haven't checked it.)

Reply:
Yeah I wasn't 100% on how to make it work well with Scala, I'll change that!